### DISCUSSION TITLE
Mobile App Development (android or IOS)

### DISCUSSION THREAD

**Discussion by shaunwbell** [NOAA-PMEL/WRC-TUG#11] [Ideas]
### Discussion:
Does anyone do mobile app development for IOS or android?  These seems like old news or circumnavigable with webapps but sometimes a utility for my group would be nice without web access (read-only view of my groups equipment databases while at sea for example).

### Potential/Suggested Speaker: 
???

**Comment by richard-parks** [comment]
I've played around with some android development but nothing too serious. I do have an android programming book at the house - if you're interested, I'll bring it in.

**Comment by shaunwbell** [comment]
Very cool.  I was curious because it came up during a NOAA Library Seminar (WeatherCitizen.org) and the product/framework [Ionic](https://ionicframework.com/) was brought up for cross platform development.  I have no experience with mobile apps so i would be curious what the community uses.

**Comment by richard-parks** [comment]
I was using Android Studio. It's based on IntelliJ IDEA and is cross platform. I haven't tried Ionic but will probably give it a shot and see how it compares. I brought the Android book in - it's on the bookshelf in my office (2106A). Anyone who's interested can drop by and borrow it at any time. I'm in no hurry to get it back - I'll eventually play around with it some more but it will be a while before that happens.

**Comment by darynwhite** [comment]
I have made websites that were mobile-focused/ready in the past, but it was over five years ago, and the game has changed immensely since then. What are you looking to build?

**Comment by shaunwbell** [comment]
The immediate use-case I can think of is a way to view (and potentially edit) select portions of our instrument deployment database without needing a network connection.  Ideally, one would download a static database copy before going to sea and then have the app display it in a particularly designed manner (on one's personal phone or mobile device) that makes sense.

I just figure a mobile app (with a prebuilt gui framework) could make quick work of this.

**Comment by darynwhite** [comment]
Ah, ok. I see the use cases for that, especially as a reference. What is running your database now? Also, would a desktop-based tool/app be more or less useful than a mobile-based one?
